### DISCUSSION TITLE
MRF Listed As Individual Links On Public Website?

### DISCUSSION THREAD

**Discussion by ghost** [CMSgov/price-transparency-guide#78] [Q&A] [answered]
Using the MRF In-Network Negotiated Rates as the file type example, a insurance carrier for a Plan/HIOS/EIN combination and based on the schema and file naming convention has 3 files for the month.

The files for insurance carrier with issuer's name "issuer abc" and the plans "healthcare-100", "gold-plated-healthcare" and "simply-plus" generated for 01/31/20222 would be

- 2022-01-31_issuer-abc_healthcare-100_in-network-rates.json
- 2022-01-31_issuer-abc_gold-plated-healthcare_in-network-rates.json
- 2022-01-31_issuer-abc_simply-plus_in-network-rates.json

In the case of most insurance carriers, the Plans/HIOS/EIN combination based on the schema and file naming convention for both individual, small group, large group and self funded could number in the 100s and 100s.

Would each of these files need to have a individual URL provided for them on the public website?
Could these files be combined into a single compressed file say 2022-01-31_issuer-abc_in-network-rates.zip?

**Comment by shaselton-usds** [comment, marked as answer]
@BobSyracuse 

You can have a single page that each file could be downloaded from if you wished. Creating a unique URL per file would be quite cumbersome. 

For example: it is perfectly acceptable to have something like `www.yourorganization.com/price-transparency/downloads` and make the files available to download from there.

**Reply by jonberkery-work** [reply]
@shaselton 
To maintain a more efficient set of in-network, out-of-network, and pharmacy files, and also improve the user\consumer experience, I am planning on providing a selection criteria screen for the consumer to use .... which would guide the user\consumer to the specific files they are searching for.

For instance I would provide a public facing front end that allows the user\consumer to choose
1. Plan Market Type - user\consumer will select "Individual" or "Group" from a drop down
2. Plan ID Type - user\consumer will select "HIOS" or "EIN" from a drop down
3. Plan ID - user\consumer will select a specific, valid, ID presented to them from a drop down - the drop down values will be based on the Plan Market Type and Plan ID Types selected above.

This reduces the amount of URLs I will manage, and presents the files dynamically.

Do you see any drawbacks with this design option?

**Reply by lizturp** [reply]
> You can have a single page that each file could be downloaded from if you wished. Creating a unique URL per file would be quite cumbersome.
> 
> For example: it is perfectly acceptable to have something like `www.yourorganization.com/price-transparency/downloads` and make the files available to download from there.

@shaselton-usds  editing because after re-reading the documentation a few times & looking at the examples again, I think I understand.  
The table of contents file (which does have a naming convention):
`www.yourorganization.com/price-transparency/downloads/2020-01-05_issuer-abc_index.json`

Can reference many plans, all of which can point to the same URL/file location or different locations.  The location could be many files zipped or just one compressed file and the naming of those files/locations don't have standards. 
`www.yourorganization.com/price-transparency/downloads/grouping-1-in-network.zip`

**Comment by jturan** [comment]
@shaselton-usds
I have another question related to this topic about where the actual files are hosted.

If an employer with a small-group plan outsources the creation of their machine-readable files to another vendor (TPA, insurer, etc.), can that vendor then host the files on their site and the employer then simply provide links to those files on the vendor's site? Or is the employer expected to host the files on their own site? I assume hosting by the vendor is acceptable, especially given the massive size of these files.

**Reply by ghost** [reply]
@jturan I am not affiliated with CMS, but what I have heard from insurers/payors, they are providing the option to host the files they generate on the vendor's website. The vendors are offering to provide a URL to the small-group to use on their site to link to the files or the option to transfer the files to the small-group to post on their public website. The Background section on Github CMS price-transparency-guide page states "through three machine-readable files posted on an internet website" with no qualifier on the hosting site being an entity like the small-group or the vendor that generates a file for the small-group.

Good question though!

**Reply by bryanbriegel** [reply]
The CFR notes "Nothing in the final rules prevents the Allowed Amount File from being hosted on a third-party website or
prevents a plan administrator from contracting with a third party to post the file. The Departments have
added text to the final rules to make clear that this flexibility exists and to provide that if a plan chooses not
to also host the file separately on its own public website, it must provide a link on its website to the location
where the file is publicly available." -- there is nothing that explicitly addresses the in-network rate or prescription drug file, though it follows if this is allowable for one machine readable file, I am not sure what would prohibit this for the others.

**Reply by nhartmanNWA** [reply]
If a group health plan uses a TPA for administration, and the group health plan does not have their own dedicated website, and they buy a fully insured product from Insurance X, and Insurance X will create and host the MRFs, does the TPA have to have pointers from their site to the Insurance X site where the files are hosted?

**Reply by shaselton-usds** [reply]
@nhartmanNWA Does the group health plan for the organization have zero presence on the internet?

@bryanbriegel You're correct that the files can be hosted in other locations with links to it from the plan.
